packages/plugin-ext/src/main/browser/languages-main.ts

```diff
@@ -105,23 +108,39 @@ export class LanguagesMainImpl implements LanguagesMain {
                 position: monaco.Position,
                 context: monaco.modes.SuggestContext,
                 token: monaco.CancellationToken): Thenable<monaco.modes.ISuggestResult> =>
-                Promise.resolve(this.proxy.$provideCompletionItems(handle, model.uri, position, context, token)).then(result => {
```

Comment:
While I'm not against this refactoring, it's completely unrelated to what you're trying to achieve in this PR.

Reply:
it's related, one can subclass `LanguagesMainImpl` and override `provideCompletionItemsImpl` to wrap it.

Reply:
Yeah, the idea was to refactor and then subclass that way I can easily wrap the requests while making minimal changes to the code.

That way I can do something like https://github.com/JPinkney/theia-plugin-extension-metrics/blob/master/src/browser/plugin-metrics-languages-main.ts#L40 and have that piece of code be called by https://github.com/eclipse-theia/theia/pull/6148/files#diff-42e21ff47aae4b2d39ad311d392f4a26R109